### PR TITLE
Deploy Buying Catalogue Config to Dynamic Environments

### PR DESCRIPTION
**Pull request by gavinwakeley1**
Task: [AB#9838](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/9838)
Story: [AB#7746](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7746)

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -310,52 +304,210 @@ jobs:
...
+  #     (source)buildType: 'specific' 
+  #     project: 'c5f97979-5b03-4d10-ba8d-871d0526b408'
+  #     pipeline: '75'
+  #     (runbranch)branchName: 'refs/heads/master'
+  #     buildVersionToDownload: 'latestFromBranch'
+  #     #allowPartiallySucceededBuilds: false
+  #     downloadType: 'single'
+  #     #artifactName: '_nhs-digital-gp-it-futures.platform-helm'
+  #     artifactName: 'build-artifact'
+  #     downloadPath: '$(System.ArtifactsDirectory)/helmArtifacts'
```
Comment (PhysicsJon) [inline azure-pipelines.yml]: Why is this block commented out?

File: generateKeyVault/keyvault_access_policy.tf [hunk 2]
```diff
@@ -188,11 +196,6 @@ resource "azurerm_key_vault_access_policy" "keyvault_current_access" {
...
+    azurerm_key_vault_access_policy.keyvault_devops_access
   ]
-
-  lifecycle {
-    ignore_changes = [
-      object_id
-    ]
-  }
-}
+}
```
Comment (PhysicsJon) [inline generateKeyVault/keyvault_access_policy.tf]: Git likes newlines at the end of files

File: generateKeyVault/keyvault_certificate.tf [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+      subject            = "CN = buyingcatalogue${local.shortEnv}.${var.region}.cloudapp.azure.com"
+      validity_in_months = 12
+    }
+  }
+
+  depends_on = [
+    azurerm_management_lock.keyvault_lock
+  ]
+}
```
Comment (PhysicsJon) [inline generateKeyVault/keyvault_certificate.tf]: EoF new line

File: generateKeyVault/keyvault_secrets.tf [hunk 4]
```diff
@@ -67,8 +60,6 @@ resource "azurerm_key_vault_secret" "kv_coreurl" {
   }
 
   depends_on = [
-    azurerm_key_vault_access_policy.keyvault_devops_access,
-    azurerm_key_vault_access_policy.keyvault_access,
-    azurerm_key_vault_access_policy.keyvault_current_access
+    azurerm_management_lock.keyvault_lock
   ]
 }
```
Comment (PhysicsJon) [inline generateKeyVault/keyvault_secrets.tf]: EoF Newline

File: terraform/application_gateways_dev.tf [hunk 5]
```diff
@@ -182,4 +185,22 @@ resource "azurerm_application_gateway" "AppGate" {
...
+      frontend_port,
+      backend_address_pool,
+      probe,
+      redirect_configuration,      
+      url_path_map,     
+      tags, # AGIC adds tags which need to be ignored. Can't seem to ignore the individual tags
+      #ssl_certificate # see issue above
+    ]
+  }
 }
```
Comment (PhysicsJon) [inline terraform/application_gateways_dev.tf]: EoF Newline

File: terraform/dns_record_dev.tf [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+resource "azurerm_dns_cname_record" "dyn_cname" {
+    count               = local.shortenv != "test" && local.shortenv != "prod" ? 1 : 0
+    name                = var.environment
+    zone_name           = data.azurerm_dns_zone.dyn_zone[0].name
+    resource_group_name = "gpitfutures-dynamic-rg"
+    ttl                 = 60
+    record              = azurerm_public_ip.PipAppGw[0].fqdn
+}
+
+
```
Comment (PhysicsJon) [inline terraform/dns_record_dev.tf]: Additional whitespace?

File: terraform/kubernetes.tf [hunk 7]
```diff
@@ -53,6 +53,10 @@ resource "azurerm_kubernetes_cluster" "aks" {
     enabled                       = "true"
   }
 
+  # identity {
+  #   type = "SystemAssigned"
+  # } 
+
```
Comment (PhysicsJon) [inline terraform/kubernetes.tf:60]: This has been added in then commented out. Is this correct?

File: terraform/kubernetes_variables.tf [hunk 8]
```diff
@@ -1,4 +1,4 @@
 locals {
-    kv_aksversion = "1.16.10"
+    kv_aksversion = "1.16.15"
     kv_aksvmsize  = "Standard_B2ms"
 }
```
Comment (PhysicsJon) [inline terraform/kubernetes_variables.tf]: EoF Newline

File: terraform/sql_databases_dev.tf [hunk 9]
```diff
@@ -1,63 +1,67 @@
...
+#   location                         = var.region
+#   server_name                      = azurerm_sql_server.sql-pri.name
+#   collation                        = local.sql_collation
+#   edition                          = local.sql_edition
+#   requested_service_objective_name = local.sql_size
+
+#   tags = {
+#     environment                    = var.environment
+#   }
+# }
```
Comment (PhysicsJon) [inline terraform/sql_databases_dev.tf]: EoF Newline

File: terraform/z_imported_secrets_core.tf [hunk 10]
```diff
@@ -25,6 +25,11 @@ data "azurerm_key_vault_secret" "sqladmins" {
...
 data "azurerm_key_vault_secret" "ssl_cert" {
-  name         = "buying-catalogue-digital-nhs-uk-star"
+  name         = "dynamic-buying-catalogue-digital-nhs-uk-star"
+  key_vault_id = data.azurerm_key_vault.keyvault_core.id
+}
+
+data "azurerm_key_vault_secret" "bjssvpn" {
+  name         = "${var.pjtcode}${local.coreEnv}bjssvpn"
   key_vault_id = data.azurerm_key_vault.keyvault_core.id
 }
```
Comment (PhysicsJon) [inline terraform/z_imported_secrets_core.tf]: EoF Newline

**Review by PhysicsJon — CHANGES_REQUESTED** [review]
Just a few small queries

File: azure-pipelines.yml (hunk 1 above)
Comment (gavinwakeley1) [reply azure-pipelines.yml]: Legacy code - deleted now

File: terraform/kubernetes.tf (hunk 7 above)
Comment (gavinwakeley1) [reply terraform/kubernetes.tf:60]: Yep, may be needed for key vault integration - comment added

**Review by PhysicsJon — APPROVED** [review]

**Pull request merged**